elk/extraction/extraction.py

```diff
@@ -275,14 +279,14 @@ def extract(
     highlight_color: str = "cyan",
     num_gpus: int = -1,
     min_gpu_mem: int | None = None,
-) -> DatasetDict:
+) -> DatasetDictWithName:
     """Extract hidden states from a model and return a `DatasetDict` containing them."""
 
     def get_splits() -> SplitDict:
         available_splits = assert_type(SplitDict, info.splits)
         train_name, val_name = select_train_val_splits(available_splits)
 
-        pretty_name = colorize(assert_type(str, info.builder_name), highlight_color)
+        pretty_name = colorize(assert_type(str, ds_name), highlight_color)
         print(
```

Comment:
previously `info.builder_name` can sometimes be none, e.g. `christykoh/imdb` has the builder name somehow.. set to None. this caused an exception for such cases.

Reply:
theres some things i tried, like setting info.builder_name to the path directly if it doesn't exist.

but theres some trippy shizzles going on (as you may have encountered before) with the builder name.

e.g. `builder.download_and_prepare` would reset the builder_name to the original info, even though we set it to something else in `_GeneratorBuilder`. This was quite annoying and hard to reason with, so i decided to just put the name in `DatasetDictWithName`